### PR TITLE
Added mixer filter for TCP connection.

### PR DESCRIPTION
**Pull request by qiwzhang**

### CODE REVIEW

File: src/envoy/mixer/tcp_filter.cc [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+      return nullptr;
+    }
+
+    Http::Mixer::TcpConfigPtr tcp_config(
+        new Http::Mixer::TcpConfig(config, server));
+    return [tcp_config](Network::FilterManager& filter_manager) -> void {
+      std::shared_ptr<Http::Mixer::TcpInstance> instance =
+          std::make_shared<Http::Mixer::TcpInstance>(tcp_config);
+      filter_manager.addReadFilter(Network::ReadFilterSharedPtr(instance));
+      filter_manager.addWriteFilter(Network::WriteFilterSharedPtr(instance));
```
Comment (douglas-reid) [inline src/envoy/mixer/tcp_filter.cc]: curious: do we want to filter both sides (read & write)?

Comment (qiwzhang) [inline src/envoy/mixer/tcp_filter.cc]: It seems that it is the only way to get write_bytes.   Other approaches may require Envoy core code change.

File: src/envoy/mixer/tcp_filter.cc [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+  }
+};
+
+}  // namespace Mixer
+}  // namespace Http
+
+namespace Server {
+namespace Configuration {
+
+class TcpMixerFilter : public NetworkFilterConfigFactory {
```
Comment (lizan) [inline src/envoy/mixer/tcp_filter.cc]: use `NamedNetworkFilterConfigFactory` instead, `NetworkFilterConfigFactory` is deprecated.

Comment (qiwzhang) [inline src/envoy/mixer/tcp_filter.cc]: Done. Thanks

**Pull request merged**